### PR TITLE
Autolathe Stuff

### PR DESCRIPTION
**Pull request by Geevies**
* Certain autolathe items can now be printed at Code Red without needing to be hacked. (though hacking allows you to access them without the level)
* Certain autolathe items have had their hacking restrictions removed, such as the industrial welder. They are now printable by default.
* Reorganized the autolathe, drastically reduced the size of the General tab, and introduced new tabs, such as Dinnerware and Armaments.
* Shotgun shells now print in piles, instead of one shell at a time. The resource cost has been increased to compensate.

https://github.com/Aurorastation/Aurora.3/assets/22774890/b17b567b-e1a8-429c-a823-14ff4f47fb00

### CODE REVIEW

**Comment by Geevies** [comment]
!review

**Comment by FluffyGhoster** [comment]
The CI tests didn't run on this PR for whatever reason, give them a kick with a commit

File: code/game/machinery/autolathe/autolathe.dm [hunk 1]
```diff
@@ -95,14 +96,28 @@
...
+		if(R.IsAbstract())
 			continue
+
+		var/is_enabled = TRUE
+		if(!hacked)
+			if(R.hack_only)
+				continue
+			if(is_on_ship && ship_security_level < R.security_level)
+				is_enabled = FALSE
+
```
Comment (Arrow768) [inline code/game/machinery/autolathe/autolathe.dm:113]: Fluffy is correct, the topic action also needs to be checked.

**Review by Arrow768 — CHANGES_REQUESTED** [review]

**Comment by Geevies** [comment]
!review

File: code/game/machinery/autolathe/autolathe.dm [hunk 2]
```diff
@@ -187,6 +202,19 @@
...
 
+		var/ship_security_level = seclevel2num(get_security_level())
+		var/is_on_ship = isStationLevel(z) // since ship security levels are global FOR NOW, we'll ignore the alert check for offship autolathes
+
+		var/is_enabled = TRUE
+		if(!hacked)
+			if(R.hack_only)
+				is_enabled = FALSE
+			else if(is_on_ship && ship_security_level < R.security_level)
+				is_enabled = FALSE
```
Comment (NonQueueingMatt) [inline code/game/machinery/autolathe/autolathe.dm:213]: this is copypasted code now, just make it into a proc

**Review by NonQueueingMatt — CHANGES_REQUESTED** [review]

**Comment by Geevies** [comment]
!review

**Review by NonQueueingMatt — APPROVED** [review]

File: code/game/machinery/autolathe/autolathe.dm [hunk 3]
```diff
@@ -187,6 +205,9 @@
 		if(!istype(R))
 			CRASH("Unknown recipe given! [R], param is [params["recipe"]].")
 
+		if(!can_print_item(R))
+			CRASH("Someone tried to print an un-enabled recipe! [R], param is [params["recipe"]].")
```
Comment (Arrow768) [inline code/game/machinery/autolathe/autolathe.dm:209]: You should print who that someone is (ckey)

File: code/game/machinery/autolathe/autolathe.dm [hunk 4]
```diff
@@ -95,14 +108,19 @@
...
-		if(R.hidden && !hacked)
+		if(R.IsAbstract())
 			continue
+
 		var/list/recipe_data = list()
 		recipe_data["name"] = R.name
 		recipe_data["recipe"] = R.type
-		recipe_data["hidden"] = R.hidden
+		recipe_data["security_level"] = R.security_level ? capitalize(num2seclevel(R.security_level)) : "None"
+		recipe_data["hidden"] = R.hack_only
```
Comment (Arrow768) [inline code/game/machinery/autolathe/autolathe.dm:121]: You should update the name of that car so ui and dm code are consistent

**Comment by Geevies** [comment]
!review

File: code/game/machinery/autolathe/autolathe.dm [hunk 5]
```diff
@@ -95,14 +108,19 @@
 	for(var/material in stored_material)
 		data["materials"] += list(list("material" = material, "stored" = stored_material[material], "max_capacity" = storage_capacity[material]))
 	data["recipes"] = list()
+
 	for(var/recipe in GET_SINGLETON_SUBTYPE_LIST(/singleton/autolathe_recipe))
 		var/singleton/autolathe_recipe/R = recipe
-		if(R.hidden && !hacked)
+		if(R.IsAbstract())
```
Comment (FluffyGhoster) [inline code/game/machinery/autolathe/autolathe.dm:114]: Use the world proc

`is_abstract(R)`

File: code/game/machinery/autolathe/designs/ammunition.dm [hunk 6]
```diff
@@ -0,0 +1,147 @@
+/singleton/autolathe_recipe/ammunition
+	name = "Abstract Ammunition"
+	category = "Ammunition"
+
+	abstract_type = /singleton/autolathe_recipe/ammunition
```
Comment (FluffyGhoster) [inline code/game/machinery/autolathe/designs/ammunition.dm:5]: `ABSTRACT_TYPE(/singleton/autolathe_recipe/ammunition)`

File: code/game/machinery/autolathe/designs/components.dm [hunk 7]
```diff
@@ -0,0 +1,55 @@
+/singleton/autolathe_recipe/components
+	name = "Abstract Components"
+	category = "Devices and Components"
+
+	abstract_type = /singleton/autolathe_recipe/components
```
Comment (FluffyGhoster) [inline code/game/machinery/autolathe/designs/components.dm:5]: `ABSTRACT_TYPE(/singleton/autolathe_recipe/components)`

File: code/game/machinery/autolathe/designs/dinnerware.dm [hunk 8]
```diff
@@ -0,0 +1,69 @@
+/singleton/autolathe_recipe/dinnerware
+	name = "Abstract Dinnerware"
+	category = "Dinnerware"
+
+	abstract_type = /singleton/autolathe_recipe/dinnerware
```
Comment (FluffyGhoster) [inline code/game/machinery/autolathe/designs/dinnerware.dm:5]: `ABSTRACT_TYPE(/singleton/autolathe_recipe/dinnerware)`

File: code/game/machinery/autolathe/designs/engineering.dm [hunk 9]
```diff
@@ -0,0 +1,55 @@
+/singleton/autolathe_recipe/engineering
+	name = "Abstract Engineering"
+	category = "Engineering"
+
+	abstract_type = /singleton/autolathe_recipe/engineering
```
Comment (FluffyGhoster) [inline code/game/machinery/autolathe/designs/engineering.dm:5]: etc

File: code/game/machinery/autolathe/designs/general.dm [hunk 10]
```diff
@@ -0,0 +1,53 @@
+/singleton/autolathe_recipe/general
+	name = "Abstract General"
+	category = "General"
+
+	abstract_type = /singleton/autolathe_recipe/general
```
Comment (FluffyGhoster) [inline code/game/machinery/autolathe/designs/general.dm:5]: etc

File: code/game/machinery/autolathe/designs/materials.dm [hunk 11]
```diff
@@ -0,0 +1,31 @@
+/singleton/autolathe_recipe/materials
+	name = "Abstract Materials"
+	category = "Materials"
+
+	abstract_type = /singleton/autolathe_recipe/materials
```
Comment (FluffyGhoster) [inline code/game/machinery/autolathe/designs/materials.dm:5]: etc

File: code/game/machinery/autolathe/designs/medical.dm [hunk 12]
```diff
@@ -0,0 +1,57 @@
+/singleton/autolathe_recipe/medical
+	name = "Abstract Medical"
+	category = "Medical"
+
+	abstract_type = /singleton/autolathe_recipe/medical
```
Comment (FluffyGhoster) [inline code/game/machinery/autolathe/designs/medical.dm:5]: etc

File: code/game/machinery/autolathe/designs/tools.dm [hunk 13]
```diff
@@ -0,0 +1,49 @@
+/singleton/autolathe_recipe/tools
+	name = "Abstract Tools"
+	category = "Tools"
+
+	abstract_type = /singleton/autolathe_recipe/tools
```
Comment (FluffyGhoster) [inline code/game/machinery/autolathe/designs/tools.dm:5]: etc

**Review by FluffyGhoster — CHANGES_REQUESTED** [review]

**Comment by Geevies** [comment]
!review

**Review by DreamySkrell — APPROVED** [review]

**Review by FluffyGhoster — APPROVED** [review]

**Pull request merged**